packages/kmod-5.10-nvidia/kmod-5.10-nvidia.spec

```diff
@@ -166,50 +184,128 @@ popd
+%{tesla_470_libdir}/libGLX_nvidia.so.%{tesla_470}
+%{tesla_470_libdir}/libGLX_nvidia.so.0
+%{tesla_470_libdir}/libEGL_nvidia.so.%{tesla_470}
+%{tesla_470_libdir}/libEGL_nvidia.so.0
+%{tesla_470_libdir}/libGLESv2_nvidia.so.%{tesla_470}
+%{tesla_470_libdir}/libGLESv2_nvidia.so.2
+%{tesla_470_libdir}/libGLESv1_CM_nvidia.so.%{tesla_470}
+%{tesla_470_libdir}/libGLESv1_CM_nvidia.so.1
+
+# NGX
+%if "%{_cross_arch}" == "x86_64"
+%{tesla_470_libdir}/libnvidia-ngx.so.%{tesla_470}
+%{tesla_470_libdir}/libnvidia-ngx.so.1
+%endif
 
 # Neither nvidia-peermem nor nvidia-drm are included in driver container images, we exclude them
 # for now, and we will add them if requested
-%exclude %{_cross_datadir}/nvidia/tesla/%{nvidia_tesla_470_version}/module-objects.d/nvidia-peermem.mod.o
-%exclude %{_cross_datadir}/nvidia/tesla/%{nvidia_tesla_470_version}/module-objects.d/nvidia-peermem.o
-%exclude %{_cross_datadir}/nvidia/tesla/%{nvidia_tesla_470_version}/module-objects.d/nvidia-drm.mod.o
-%exclude %{_cross_datadir}/nvidia/tesla/%{nvidia_tesla_470_version}/module-objects.d/nvidia-drm.o
-%exclude %{_cross_libexecdir}/nvidia/tesla/bin/%{nvidia_tesla_470_version}/nvidia-cuda-mps-control
-%exclude %{_cross_libexecdir}/nvidia/tesla/bin/%{nvidia_tesla_470_version}/nvidia-cuda-mps-server
+%exclude %{_cross_datadir}/nvidia/tesla/%{tesla_470}/module-objects.d/nvidia-peermem.mod.o
+%exclude %{_cross_datadir}/nvidia/tesla/%{tesla_470}/module-objects.d/nvidia-peermem.o
+%exclude %{_cross_datadir}/nvidia/tesla/%{tesla_470}/module-objects.d/nvidia-drm.mod.o
+%exclude %{_cross_datadir}/nvidia/tesla/%{tesla_470}/module-objects.d/nvidia-drm.o
+%exclude %{_cross_libexecdir}/nvidia/tesla/bin/%{tesla_470}/nvidia-cuda-mps-control
+%exclude %{_cross_libexecdir}/nvidia/tesla/bin/%{tesla_470}/nvidia-cuda-mps-server
 %if "%{_cross_arch}" == "x86_64"
-%exclude %{_cross_libexecdir}/nvidia/tesla/bin/%{nvidia_tesla_470_version}/nvidia-ngx-updater
+%exclude %{_cross_libexecdir}/nvidia/tesla/bin/%{tesla_470}/nvidia-ngx-updater
 %endif
+
+# None of these libraries are required by libnvidia-container, so they
+# won't be used by a containerized workload
+%exclude %{tesla_470_libdir}/libGLX.so.0
+%exclude %{tesla_470_libdir}/libGLdispatch.so.0
+%exclude %{tesla_470_libdir}/libOpenCL.so.1.0.0
+%exclude %{tesla_470_libdir}/libOpenGL.so.0
+%exclude %{tesla_470_libdir}/libglxserver_nvidia.so.470.82.01
+%exclude %{tesla_470_libdir}/libnvidia-allocator.so.470.82.01
+%exclude %{tesla_470_libdir}/libnvidia-egl-wayland.so.1.1.7
+%exclude %{tesla_470_libdir}/libnvidia-gtk2.so.470.82.01
+%exclude %{tesla_470_libdir}/libnvidia-gtk3.so.470.82.01
+%exclude %{tesla_470_libdir}/libnvidia-nvvm.so.4.0.0
+%exclude %{tesla_470_libdir}/libnvidia-opencl.so.470.82.01
+%exclude %{tesla_470_libdir}/libnvidia-vulkan-producer.so.470.82.01
+%exclude %{tesla_470_libdir}/nvidia_drv.so
+%exclude %{tesla_470_libdir}/libOpenCL.so.1
+%exclude %{tesla_470_libdir}/libnvidia-egl-wayland.so.1
```

Comment:
Except (possibly) for OpenCL, these look OK to exclude - `nvidia_drv` is for the X server, and `libnvidia-egl-wayland` is Wayland related.